### PR TITLE
Feature/provide ef framwork adapter

### PR DESCRIPTION
**Pull request by ChangeTheCode**
- implementation based on ef framework core 
- support ef framework adapter for casbin 
- provide unit tests

### CODE REVIEW

**Comment by hsluoyz** [comment]
@huazhikui @xcaptain @Sagilio plz review.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+language: csharp
+mono: none
+solution: EF-Adapter.sln
+dist: xenial
+dotnet: 
+    - 3.1
```
Comment (xcaptain) [inline .travis.yml]: it's .net core 3.1, this project is intended to support .net framework

**Review by xcaptain — COMMENTED** [review]
Have no .net framework test environment so can't help test, but the dependencies are for .net core and efcore, the latest .net framework version is 4.8

File: EF-Adapter/CasbinModel.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        private readonly ConfigurationRegistrar _CasbinModelConfig;
+        public virtual DbSet<CasbinRule<TKey>> CasbinRule { get; set; }
+        
+
+        public CasbinDbContext(string databaseConnectionString)
+            : this(databaseConnectionString, new ValidateDbInitializer<CasbinDbContext<TKey>>(), false)
+        {
+        }
+
+        public CasbinDbContext(string databaseConnectionString, EntityTypeConfiguration<CasbinDbContext<TKey>> entityType)
```
Comment (sagilio) [inline EF-Adapter/CasbinModel.cs]: `entityType` is never used.

File: EF-Adapter/CasbinModel.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        public CasbinDbContext(string databaseConnectionString, IDatabaseInitializer<CasbinDbContext<TKey>> initializer, bool forceInit = true) 
+            : base(databaseConnectionString)
+        {
+            Database.SetInitializer(initializer);
+            Database.Initialize(forceInit);
+        }
+
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            if (_CasbinModelConfig != null)
```
Comment (sagilio) [inline EF-Adapter/CasbinModel.cs]: `_CasbinModelConfig` can not be initialized by ctor.

File: EF-Adapter/EF-Adapter.csproj [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <RootNamespace>Casbin.NET.Adapter.EF</RootNamespace>
+    <PackageId>Casbin.NET.Adapter.EF</PackageId>
+    <Version>1.2.1</Version>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Casbin.NET" Version="1.3.0" />
+    <PackageReference Include="EntityFramework" Version="6.1.3" />
```
Comment (sagilio) [inline EF-Adapter/EF-Adapter.csproj]: Why not use the new version like `v6.4.4`?

**Review by sagilio — CHANGES_REQUESTED** [review]
Thank you, it looks good, but there are a few questions.
Would you like to add the unit test in a later commit?

File: EF-Adapter/EF-Adapter.csproj (hunk 4 above)
Comment (ChangeTheCode) [reply EF-Adapter/EF-Adapter.csproj]: i thought it was in time of implementation the newest but I will update to v.6.4.4

File: .travis.yml (hunk 1 above)
Comment (hsluoyz) [reply .travis.yml]: @ChangeTheCode

Comment (ChangeTheCode) [reply .travis.yml]: I'm sorry. I'm not familiar with travis.yml. I take that from the EFCore-Adapter and did not adjust it. 
Can someone help me out here ?

Comment (hsluoyz) [reply .travis.yml]: @Sagilio can you help on it?

Comment (sagilio) [reply .travis.yml]: Travis does not support the .net framework well, we can use Github Actions or Azure Pipelines. Now, Casbin.NET and EFCore adapter project use the Github Actions mainly. 
You can delete the `.travis.yml` file directly.

**Review by sagilio — APPROVED** [review]

**Comment by sagilio** [comment]
Thank you, the unit test is passed, and could you change the EF-Adapter target frameworks to :
```
<TargetFrameworks>net45;net461;netstandard2.1</TargetFrameworks>
```
and the EF-Adapter.UnitTest target frameworks to :
```
<TargetFrameworks>net452;net461;netcoreapp3.1</TargetFrameworks>
```
I have tried these changes are ok.

**Comment by hsluoyz** [comment]
@ChangeTheCode any update?

**Comment by hsluoyz** [comment]
@Sagilio merged for now, plz make another PR to update TargetFrameworks.

**Pull request merged**